### PR TITLE
tool: add skopeo, crictl into dev dockerfile

### PR DESCRIPTION
**Pull request by liangzhou121**
Signed-off-by: Zhou Liang <liang.a.zhou@linux.alibaba.com>

### CODE REVIEW

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 1]
```diff
@@ -1,3 +1,68 @@
+# The first stage
+FROM ubuntu:18.04 AS src
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:2]: Line 1-33 repeats with Line 69-97. Please improve it.

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 2]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
...
+    rm -rf cri-containerd-cni-1.5.3-linux-amd64.tar.gz etc opt usr
+
+# configure containerd
+RUN containerd config default > config.toml && \
+    sed -i "s#k8s.gcr.io/pause:3.1#registry.cn-hangzhou.aliyuncs.com/acs/pause-amd64:3.1#g" config.toml && \
+    sed -i 's/\(snapshotter = \)"overlayfs"/\1"native"/' config.toml
+
+# build skopeo
+RUN git clone https://github.com/containers/skopeo $GOPATH/src/github.com/containers/skopeo && \
+    apt-get install -y libdevmapper-dev && apt-get install -y build-essential nghttp2 libnghttp2-dev libssl-dev && apt-get install -y libfftw3-dev libsndfile1-dev libgpgme-dev && \
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:55]: Please change to `apt-get install -y libdevmapper-dev  build-essential nghttp2 libnghttp2-dev libssl-dev libfftw3-dev libsndfile1-dev libgpgme-dev`

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 3]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
...
+RUN echo "runtime-endpoint: unix:///run/containerd/containerd.sock\nimage-endpoint: unix:///run/containerd/containerd.sock\ntimeout: 2\ndebug: true" > /etc/crictl.yaml
+
+# configue ocicrypt keyprovider
+RUN mkdir -p /etc/containerd/ocicrypt/keys
+
+# configure containerd
+RUN sed -i "s#k8s.gcr.io/pause:3.5#registry.cn-hangzhou.aliyuncs.com/acs/pause-amd64:3.1#g" config.toml && \
+    cp -f config.toml /etc/containerd
+
+
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:71]: Please delete this empty line.

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 4]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
...
+    cd $GOPATH/src/github.com/containers/skopeo && make bin/skopeo
+
+# install skopeo
+RUN cp -f $GOPATH/src/github.com/containers/skopeo/bin/skopeo /usr/local/bin/
+
+# configue crictl
+RUN echo "runtime-endpoint: unix:///run/containerd/containerd.sock\nimage-endpoint: unix:///run/containerd/containerd.sock\ntimeout: 2\ndebug: true" > /etc/crictl.yaml
+
+# configue ocicrypt keyprovider
+RUN mkdir -p /etc/containerd/ocicrypt/keys
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:65]: The step of `configue ocicrypt keyprovider` just create a new directory?

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 5]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
...
+
+# configure containerd
+RUN sed -i "s#k8s.gcr.io/pause:3.5#registry.cn-hangzhou.aliyuncs.com/acs/pause-amd64:3.1#g" config.toml && \
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:68]: The stop of `configure containerd` has been done in the previous step.

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 6]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
...
+    sed -i "s#k8s.gcr.io/pause:3.1#registry.cn-hangzhou.aliyuncs.com/acs/pause-amd64:3.1#g" config.toml && \
+    sed -i 's/\(snapshotter = \)"overlayfs"/\1"native"/' config.toml
+
+# build skopeo
+RUN git clone https://github.com/containers/skopeo $GOPATH/src/github.com/containers/skopeo && \
+    apt-get install -y libdevmapper-dev && apt-get install -y build-essential nghttp2 libnghttp2-dev libssl-dev && apt-get install -y libfftw3-dev libsndfile1-dev libgpgme-dev && \
+    cd $GOPATH/src/github.com/containers/skopeo && make bin/skopeo
+
+# install skopeo
+RUN cp -f $GOPATH/src/github.com/containers/skopeo/bin/skopeo /usr/local/bin/
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:59]: The `install skopeo ` step can combine with `build skopeo` step

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 [hunk 7]
```diff
@@ -35,6 +35,42 @@ ENV GO111MODULE  on
 RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path
 ENV PATH         /root/.cargo/bin:$PATH
 
+
```
Comment (haosanzi) [inline tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:38]: Please delete this empty line

File: tools/docker/Dockerfile-inclavare-dev-ubuntu18.04 (hunk 4 above)
Comment (liangzhou121) [reply tools/docker/Dockerfile-inclavare-dev-ubuntu18.04:65]: Yes, this empty directly is needed or the image pulling step will exit with error directly.

**Review by haosanzi — APPROVED** [review]

**Pull request merged**